src/python/pants/engine/internals/parser.py

```diff
@@ -131,13 +170,15 @@ def __call__(self, **kwargs: Any) -> TargetAdaptor:
 
         symbols: dict[str, Any] = {
             **object_aliases.objects,
-            "build_file_dir": lambda: PurePath(parse_state.rel_path()),
+            "build_file_dir": lambda: PurePath(parse_state.filepath()).parent,
```

Comment:
I'm confused about the rel_path/filepath name and semantics change. Why did this need to change?

Reply:
I wanted the `filepath` to the BUILD file being parsed, for any error messages to be precise. See the call to `BuildFileDependencyRuleParserState.set_dependency_rules()`, it's the `build_file: str` arg. So rather than trimming it going into `ParseState` I opted to preserve it, and only trim it in the property offering the `rel_path`.

So the semantics change should only be internal to the parser and not leak out further than to the oracle (if it does, it's a bug).

Reply:
Got it, makes sense. This is only confusing to me reading this PR, it will not be confusing to future readers of the code.